tests/alembic_migrations/test_warnings.py

```diff
@@ -0,0 +1,21 @@
+import warnings
+import pytest
+
+from sqlalchemy.exc import SAWarning
+from .alembicverify_util import prepare_schema_from_migrations
+
+
+@pytest.mark.usefixtures("new_db_left")
+def test_warnings(
+    uri_left, alembic_config_left, db_setup, env_config
```

Comment:
I must have missed the point at which these fixtures were introduced. After a read of `conftest.py` it's not overwhelmingly obvious to me exactly what they do. Are the left/right fixtures db URIs for pre/post migration?

Reply:
Fixtures `new_db_left`, `uri_left`, `alembic_config_left` are provided by package [`alembic-verify`](https://alembic-verify.readthedocs.io/en/latest/), which simplifies testing Alembic migrations.

As useful as `alembic-verify` is, it didn't quite do the whole job, so I also created our own module `alembicverify_util`, which is a reworking of the `alembicverify.util` module. What it does is described in our module. Fixtures `db_setup` and `env_config` provide test-specific information to the the tests which call call functions in `alembicverify_util`.

Obviously this isn't clear enough in the code and I will document it there.